File: 1760/Solution.java
class Solution {
    public int minimumSize(int[] nums, int maxOperations) {
        int l = 1, r = 1_000_000_000;
        while(l <= r){
            int m = (r + l) / 2;
            int count = 0;
            for(int i : nums) count += (i -1)/ m;
            if(count > maxOperations){
                l = m + 1;
            }else{
                r = m - 1;
            }
        }
        return l;
    }
}

File: 1760/Solution.go
func minimumSize(nums []int, maxOperations int) int {
    l, r := 1, 1000000000
    for l <= r {
        m := (r + l) / 2
        k := 0
        for _, i := range nums {
            k += (i - 1) / m
        }
        if k <= maxOperations {
            r = m - 1
        }else {
            l = m + 1
        }
    }
    return l
}
